src/core/logic.py

```diff
@@ -37,79 +38,101 @@
 
 def send_reset_token(request, reset_token):
     core_reset_password_url = request.site_type.site_url(
-        reverse(
+        reverse_with_next(
             'core_reset_password',
             kwargs={'token': reset_token.token},
+            request=request,
         )
     )
```

Comment:
As discussed, since `request.site_type.site_url` now takes in an optional `query` parameter, we want to avoid encoding the querystring onto the `path` parameter ahead of time. Instead, we can pass in the query elements to be encoded separately and delegete URL encoding to `site_url`. We could either use a regular dictionary or a mutable [`Querydict`](https://docs.djangoproject.com/en/4.2/_modules/django/http/request/#QueryDict)